File: Algorithms/p141_Linked_List_Cycle/p141_Linked_List_Cycle_testcase.cpp
int main(){
    ListNode* t1 = new ListNode(1);
    ListNode* t2 = new ListNode(2);
    ListNode* t3 = new ListNode(3);
    ListNode* t4 = new ListNode(4);
    ListNode* t5 = new ListNode(5);
    t1->next = t2;
    t2->next = t3;
    t3->next = t4;
    t4->next = t5;
    t5->next = t3;

    Solution s;
    cout<<s.hasCycle(0)<<endl;
    cout<<s.hasCycle(t1)<<endl;

    return 0;
}

File: Algorithms/p141_Linked_List_Cycle/p141_Linked_List_Cycle_testcase.py
t1 = ListNode(1)
t2 = ListNode(2)
t3 = ListNode(3)
t4 = ListNode(4)
t5 = ListNode(5)
t1.next = t2
t2.next = t3
t3.next = t4
t4.next = t5
t5.next = t3

s = Solution()
print(s.hasCycle(None))
print(s.hasCycle(t1))
